runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSourceWrapper.java

```diff
@@ -67,40 +78,51 @@ public String getName() {
   @Override
   public void run(SourceContext<WindowedValue<T>> ctx) throws Exception {
```

Comment:
I don't fully understand. `SourceContext` is part of Flink and even if we inserted the `PipelineOptions` there, we would have to serialize it, right? We could provide a special SourceContext for Beam. That would still require us to do some casting in the run method or to provide a special SourceFunction interface for Beam. Not sure if that is easier.